### ISSUE TITLE
Bug: tools/config-rule ignores `oneOf` branches when generating eslint-fuzzer configs

### ISSUE THREAD

**Issue by techmannih** [eslint/eslint#20909]
### Environment

Node version: v22.18.0  
npm version: v11.6.0  
Local ESLint version: HEAD (repo checkout; `package.json` version `10.4.0`)  
Global ESLint version: Not found  
Operating System: Linux 6.8.0-106-generic 

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
// No ESLint config is required for this reproduction.
// The issue is in ESLint's internal config generator used by the fuzzer.
```
</details>

```js
const { generateConfigsFromSchema } = require("./tools/config-rule");
const builtInRules = require("./lib/rules");

for (const id of ["max-params", "generator-star-spacing", "no-unused-vars"]) {
    const rule = builtInRules.get(id);
    console.log(id, generateConfigsFromSchema(rule.meta.schema));
}
```

### What did you expect to happen?

I expected `generateConfigsFromSchema()` to generate valid rule configurations that cover `oneOf` branches instead of reducing these rules to severity-only configs.

For example:

- `max-params` defines a `oneOf` schema with integer and object branches.
- `generator-star-spacing` defines `oneOf` branches for string and object forms.
- `no-unused-vars` also defines optioned schema branches.

Because `tools/eslint-fuzzer.js` uses `createCoreRuleConfigs(true)`, I expected the generated configs to include at least some non-default option variants so the fuzzer can exercise those code paths.

### What actually happened?

The generator returned only severity-only configs for all tested `oneOf` rules:

```txt
max-params [2]
generator-star-spacing [2]
no-unused-vars [2]
```

This means the fuzzer appears to miss option-specific paths for many rules whose schemas rely on oneOf.

### Link to Minimal Reproducible Example

https://github.com/techmannih/eslint.git

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [x] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by DMartens** [comment]
We are already working on this and other enhancements in #20788.
